### PR TITLE
feat(reservation): add option to decrease amount when booking

### PR DESCRIPTION
**Pull request by hututudor**
https://etvas.atlassian.net/browse/ETV-11967

### CODE REVIEW

File: src/helpers/reservations.ts [hunk 1]
```diff
@@ -614,6 +619,11 @@ const bookReservation = async (person, reservation, increaseAmount) => {
       );
   }
 
+  if (decreaseAmount) {
+    additionalAmount =
+      -1 * (1 + Math.random() * reservation.amount.value * 0.98);
```
Comment (ohnu) [inline src/helpers/reservations.ts]: ar fi bine sa fie predictibila suma in loc de random ca sa putem scrie teste e2e cu un assert cunoscut

Comment (ohnu) [reply src/helpers/reservations.ts]: in ambele cazuri (si increase si decrease)

Comment (ohnu) [reply src/helpers/reservations.ts]: ex: increase / decrease by 50% or something like that

Comment (hututudor) [reply src/helpers/reservations.ts]: https://github.com/etvascom/mock-solaris/pull/30/commits/aec5b784f77d1927de34ae70f0eee1e62e92f917

**Review by ohnu — APPROVED** [review]

**Pull request merged**